monitor/metrics/store.go

```diff
@@ -75,9 +76,16 @@ func (mtrs *Store) LatestValid(name string) []*api.Metric {
 		return []*api.Metric{}
 	}
 
-	metrics := make([]*api.Metric, 0, len(byPeer))
-	for _, window := range byPeer {
-		m, err := window.Latest()
+	peers := []peer.ID{}
+	for p := range byPeer {
+		peers = append(peers, p)
+	}
+	sortedPeers := peer.IDSlice(peers)
+	sort.Stable(sortedPeers)
+
+	metrics := make([]*api.Metric, 0, len(peers))
+	for _, p := range peers {
+		m, err := byPeer[p].Latest()
```

Comment:
let's do this cleanly:

* create MetricSlice type
* Implement the sort.Interface
* Use that type here and call sort.Stable(metrics)
* I'm not sure if better to sort by peer or by expiry time (perhaps first by peer, second by expire time).

This will save a bunch of allocations, make everything cleaner and more readable.